documentation/src/main/asciidoc/chapter-14-third-party-api-integration.asciidoc

```diff
@@ -0,0 +1,43 @@
+[[third-party-api-integration]]
+== Third-party API integration
+
+[[non-shipped-annotations]]
+=== Non-shipped annotations
+
+There are various use-cases you must resolve ambiguity for MapStruct to use a correct piece of code.
+However, the primary goal of MapStruct is to focus on bean mapping without polluting the entity code.
+For that reason, MapStruct is flexible enough to interact with already defined annotations from third-party libraries.
+The requirement to enable this behavior is to match the _name_ of such annotation.
+Hence, we say that annotation can be _from any package_.
+
+For example, MapStruct searches for the annotation _named_ `@ConstructorProperties`.
+As long as there already exists `java.beans.ConstructorProperties` annotation from Java SE, the idea is to reuse such annotation.
```

Comment:
I would not suggest reusing the `java.beans.ConstructorProperties`, especially on Java 9+ and the fact that the `ConstructorProperties` is part of the `java.desktop` module.

Reply:
Thanks for the review. My mind is stuck in Java 8. Well, the idea of this sentence is to give an example of reusing the already existing annotation, not suggesting particularly this one. I remove the sentence for now to avoid the confusion. Do you have an idea of an existing annotation to demonstrate such usage or a better way to rephrase the sentence?

Reply:
Missed your question. Unfortunately I don't have an idea